pkg/smartcontract/manifest/standard/comply.go

```diff
@@ -37,24 +38,10 @@ func Check(m *manifest.Manifest, standards ...string) error {
 
 // Comply if m has all methods and event from st manifest and they have the same signature.
 // Parameter names are ignored.
-func Comply(m, st *manifest.Manifest) error {
+func Comply(m *manifest.Manifest, st *Standard) error {
```

Comment:
Are you sure that we need to provide a standard to `Comply`? The manifest itself has the list of supported standards, and I think that we only need to check the compliance with those standards which are in the `manifest.SupportedStandards` list.

Or `standard.Check` function can be refactored a bit to check the compliance only for `manifest.SupportedStandards` methods because the only  place where `standard.Check` is used (not in tests)  is in the compiler:
```
if err := standard.Check(m, o.ContractSupportedStandards...); err != nil {
```
And by this moment `m` already has these supported standards in the `m.SupportedStandards` list.

Reply:
I'd leave it as it is, because this is actually a generic mechanism for checking if contract satisfies some interface, not necessarily `NEP*` standard which is used in manifest. I can see us extending this set of checks in future for compiler.
Something like `NEP11-payable` immediately comes to mind, especially if we move is out of `NEP-11`.